src/transformers/models/led/modeling_tf_led.py

```diff
@@ -1818,24 +1840,34 @@ def call(
         hidden_states = inputs["inputs_embeds"]
 
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask = None
-        if input_shape[-1] > 1:
-            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
-        else:
-            combined_attention_mask = _expand_mask(
+        combined_attention_mask = tf.cond(
```

Comment:
Let's solve this exactly how we solved it for TFBart. TFLed should use the exact same code as TFBart, here: https://github.com/huggingface/transformers/blob/02e05fb0a532e572b56ba75dad6ba3db625bbdeb/src/transformers/models/bart/modeling_tf_bart.py#L898

-> think the Bart way is much cleaner

Reply:
Same reason, `input_shape[-1] > 1` is a tensor and we are not allowed to use a tensor as a Python boolean :(

Reply:
So it doesn't work in TFBart either?

Reply:
It works in BART because we don't ask Autograph to go through all the model with `@tf.function` nevertheless having `self.led._set_save_spec(inputs=self.serving.input_signature)` as the side effect to be understood by the TF tracer like if there was `tf.function(TFLEDModel.call)`. Then, for this case, Autograph is running and then fails with:
```
tensorflow.python.framework.errors_impl.OperatorNotAllowedInGraphError: using a `tf.Tensor` as a Python `bool` is not allowed: AutoGraph did convert this function. This might indicate you are trying to use an unsupported feature.
```